### PR TITLE
Added custom standards

### PR DESCRIPTION
**Pull request by ha17**
To Test: pull branch and run `bin/phpunit`

--

1) Disabled camelCase method name requirement for models matching `*Model.php`
2) For `Route()` attributes, the constructor argument should strictly-match (`===`)  the variable name.

```
    public function incoming(
        #[Route(new IntPipe(), 'fooBaz')] int $foo_baz,
        #[Route(new IntPipe(), 'fooBar')] int $fooBar,
    ) {
    }
```

The first one will error, the second one is valid.

**Added unit tests.**
Once approved will tag new release to bring into other repositories.

### CODE REVIEW

**Review by pcrist — APPROVED** [review]
I'm somewhat removed from the coding standards internals but this looks ok to me.

File: src/PDGA/Sniffs/Attributes/MatchingNamesSniff.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+
+        $stackPtr = $phpcsFile->findNext(T_STRING, $stackPtr);
+        $tokenInfo = $tokens[$stackPtr];
+
+        // Only for Route Attributes,
+        // so return if it not one.
+        if ($tokenInfo['content'] !== 'Route') {
```
[the remark this answers is not in the document]
Comment (ha17) [reply src/PDGA/Sniffs/Attributes/MatchingNamesSniff.php:26]: I made a note of this and will examine when I have more time. I would like to test but the way it parses into File and uses itself to process that file has me a little thrown off.

**Comment by ha17** [comment]
As an FYI, here's the system in action, via a testing version of `coding-standards`

image

image

**Review by drew-acheson — APPROVED** [review]

**Review by timallen527 — APPROVED** [review]

**Comment by ha17** [comment]
Re-requesting reviews due to unit tests now in the repo.  Slight change to the main sniff file as well, just a little refactoring.

File: Tests/Sniffs/Attributes/RouteAttributeNamesMustMatchTest.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    private RunPhpCs $runPhpCs;
+
+    public function setUp(): void
+    {
+        $this->runPhpCs = new RunPhpCs();
+
+        parent::setUp();
+    }
+
+    // If a
```
Comment (pcrist) [inline Tests/Sniffs/Attributes/RouteAttributeNamesMustMatchTest.php:23]: Short comment!

**Review by pcrist — APPROVED** [review]

**Pull request merged**